include/eve/module/real/proba/function/regular/generic/uniform_discrete_distribution.hpp

```diff
@@ -0,0 +1,289 @@
+      EVE_ASSERT(all(is_flint(b)), "b must be flint");
+      EVE_ASSERT(all(a < b)       , "a must be stricty less than b");
+    }
+
+    template < typename G, typename R = value_type> auto operator()(G & gen, as_<R> const & )
+      requires scalar_value<value_type>
+    {
+      return floor(fma(detail::urg01(gen, as<R>()), n, a));
+    }
+
+    value_type a;
+    value_type b;
+    value_type n;
+  };
+
+
+  template < floating_real_value T>
+  struct uniform_discrete_distribution<callable_zero_, callable_one_, T>// : uniform_discrete_distribution_base < T, T, T>
+  {
+    using is_distribution_t = void;
+    using discrete_t = void;
+    using a_type = callable_zero_;
+    using b_type = callable_one_;
+    using value_type = T;
+    using elt_t  = element_type_t<T>;
+
+    template < typename G, typename R = value_type> auto operator()(G & gen, as_<R> const & )
+      requires scalar_value<value_type>
+    {
+      return floor(2*detail::urg01(gen, as<R>()));
+    }
+
+    constexpr uniform_discrete_distribution() { }
+    constexpr uniform_discrete_distribution( as_<T> const&) { }
+
+  };
+
+  template<floating_real_value T, floating_real_value U>  uniform_discrete_distribution(T,U) -> uniform_discrete_distribution<T,U>;
+  template<floating_real_value T, floating_real_value U, floating_real_value V>  uniform_discrete_distribution(T,U,V) -> uniform_discrete_distribution<T,U, V>;
+  template<floating_real_value T, floating_real_value U, floating_real_value V>  uniform_discrete_distribution(T,U,as_<V>) -> uniform_discrete_distribution<T,U,V>;
+
+  template<integral_value T, integral_value U>  uniform_discrete_distribution(T,U) -> uniform_discrete_distribution<T,U>;
+  template<integral_value T, integral_value U, floating_real_value V>  uniform_discrete_distribution(T,U,V) -> uniform_discrete_distribution<T,U, V>;
+  template<integral_value T, integral_value U, floating_real_value V>  uniform_discrete_distribution(T,U,as_<V>) -> uniform_discrete_distribution<T,U,V>;
+
+  template<typename T>  uniform_discrete_distribution(T const&) -> uniform_discrete_distribution<callable_zero_, callable_one_, T>;
+  template<typename T>  uniform_discrete_distribution(as_<T> const&) -> uniform_discrete_distribution<callable_zero_, callable_one_, as_<T>>;
+
+  template<value T>
+  inline constexpr auto uniform_discrete_distribution_01 = uniform_discrete_distribution<callable_zero_, callable_one_, T>();
+
+  namespace detail
+  {
+    //////////////////////////////////////////////////////
+    /// cdf discrete
+    template<typename T, typename U, floating_value V, typename I = V>
+    EVE_FORCEINLINE  auto cdf_(EVE_SUPPORTS(cpu_)
+                              , uniform_discrete_distribution<T, U, I> const & d
+                              , V const &x ) noexcept
+     requires compatible_values<V, typename uniform_discrete_distribution<T, U, I>::value_type>
+     {
+      if constexpr(value<T> && value<U>)
```

Comment:
I don't understand the logic. T and U are of course value, when are we supposed to go the else ?

Reply:
no they can be callable one or zero